src/Induction/WellFounded.agda

```diff
@@ -112,6 +113,24 @@ module FixPoint
   unfold-wfRec : ∀ {x} → wfRec P f x ≡ f x (λ y _ → wfRec P f y)
   unfold-wfRec {x} = f-ext x wfRecBuilder-wfRec
 
+------------------------------------------------------------------------
+-- Well-founded relations are asymmetric and irreflexive.
+
+module _ {_<_ : Rel A r} where
+  Acc-asymm : (x : A) → (Acc _<_ x) → (y : A) → x < y → ¬ (y < x)
+  Acc-asymm = Some.wfRec _ λ x hx y x<y y<x → hx y y<x x y<x x<y
+
+  Wf-asymm : WellFounded _<_ → Asymmetric _<_
+  Wf-asymm wf x<y y<x = Acc-asymm _ (wf _) _ x<y y<x
+
+  Acc-irrefl : {_≈_ : Rel A ℓ} → Symmetric _≈_ → _<_ Respects₂ _≈_ → 
+               (x : A) → Acc _<_ x →  (y : A) → x ≈ y → ¬ (x < y)
+  Acc-irrefl s r x p y x≈y x<y = 
+    Acc-asymm x p y x<y (proj₂ r x≈y (proj₁ r (s x≈y) x<y))
+
+  Wf-irrefl : WellFounded _<_ → {_≈_ : Rel A ℓ} → Symmetric _≈_ → 
+              _<_ Respects₂ _≈_ → Irreflexive _≈_ _<_
+  Wf-irrefl wf s r = Acc-irrefl s r _ (wf _) _
```

Comment:
These proofs are (redundant) consequences of the preceding two, coupled with `Relation.Binary.Consequences.asym⇒irr`. Indeed, they don't as such depend on `Acc` or `WellFounded`, so it's not obvious that they belong here? 

Or if you think they do, then please refactor to use the above lemma.

Reply:
So, I spent some time playing with this, and came to the following compromise position, for your consideration:
* do you need `acc-irrefl` as a stand-alone result? if so, then there's nothing further to do besides fix the `CHANGELOG`
* if not, then `acc-asym` and `wf-asym` can work as they stand, and `wf-irrefl` can simply delegate to `wf-asym` and `asym⇒irr`

I prefer the second choice, but you may want to retain the more 'local' property as well.

Reply:
I fixed the changelog, and simplified the proof of `wf-irrefl` as you suggested removing `acc-irrefl`. Thank you for the comments again.